source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -600,10 +600,14 @@ inline void ComponentInterface<NodeT>::add_input(
       throw exceptions::AddSignalException("Failed to add input '" + signal_name + "': Input already exists");
     }
     std::string topic_name = default_topic.empty() ? "~/" + parsed_signal_name : default_topic;
-    this->add_parameter(
-        parsed_signal_name + "_topic", topic_name, "Output topic name of signal '" + parsed_signal_name + "'",
-        fixed_topic
-    );
+    auto parameter_name = parsed_signal_name + "_topic";
```

Comment:
This addition makes sense. It also doesn't counteract the discussion we had about adding and removing signals - since you are already checking for the existence of the parameter it will be easier to integrate that planned feature in a future PR